packages/typespec-client-generator-core/src/package.ts

```diff
@@ -46,6 +52,47 @@ export function createSdkPackage<TServiceOperation extends SdkServiceOperation>(
   return diagnostics.wrap(sdkPackage);
 }
 
+function createClients<TServiceOperation extends SdkServiceOperation>(
```

Comment:
Can we move this client hierarchy build process into cache builder section. I prefer after we get the client hierarchy info, all the following calculation does not have special logic for multi-service scenario. The only exception could be API version cache builder.